backstory/src/test/java/com/google/sps/perspective/data/MockPerspectiveAPIFactory.java

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.perspective.data;
+
+import static org.mockito.Mockito.*;
+
+import au.com.origma.perspectiveapi.v1alpha1.PerspectiveAPI;
+import au.com.origma.perspectiveapi.v1alpha1.models.AnalyzeCommentRequest;
+import au.com.origma.perspectiveapi.v1alpha1.models.AnalyzeCommentResponse;
+import au.com.origma.perspectiveapi.v1alpha1.models.AttributeScore;
+import au.com.origma.perspectiveapi.v1alpha1.models.AttributeType;
+import au.com.origma.perspectiveapi.v1alpha1.models.Score;
+import com.google.sps.perspective.PerspectiveStoryAnalysisManager;
+import com.google.sps.perspective.data.APINotAvailableException;
+import com.google.sps.perspective.data.NoAppropriateStoryException;
+import com.google.sps.perspective.data.StoryDecision;
+import java.util.Map;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/** 
+ * An immutable factory that takes in a map of desired scores
+ * and can produce instances that return those values.
+ */
+public class MockPerspectiveAPIFactory implements PerspectiveAPIFactory {
+  
+  /** the default value that a mockAPI will return for attribute score (not a valid value) */
+  private static final int DEFAULT_ATTRIBUTE_SCORE = -1;
+
+  /** a map of the scores the API should return */
+  private Map<AttributeType, Float> cannedScores;
+
+  /** the one instance of this API to return */
+  private final PerspectiveAPI perspectiveAPI; 
+
+  /** 
+   * Constructs a MockPerspective APIFactory object which will return certain scores
+   * (those passed in through desiredScores) and DEFAULT_ATTRIBUTE_SCORE if no scores 
+   * passed in.
+   *
+   * @param desiredScores a map of the scores you want the mockAPI to return 
+   *    (if you don't set any, the mock version of this API will always return DEFAULT_ATTRIBUTE_SCORE)
+   */
+  public MockPerspectiveAPIFactory(Map<AttributeType, Float> desiredScores) {
+    cannedScores = desiredScores;
+    perspectiveAPI = createMockAPI(desiredScores);
+  }
+
+  /** 
+   * Generates a mock version of the PerspectiveAPI which will return the scores
+   * set in the constructor or which returns a default value when there is no desired score
+   *
+   * @return an instance of a mock Perspective API 
+   */
+  public PerspectiveAPI newInstance() {
+    return perspectiveAPI;
```

Comment:
The implication of `newInstance()` is that the caller will get a new instance. This could be resolved here by instead doing: `return createMockAPI(desiredScores);`

Reply:
Ok I will make that change. Thank you so much!

Reply:
Done